### DISCUSSION TITLE
ImageMagick does not apply formatting to text in a PNG image Empty ImageMagick does not apply formatting to text in a PNG image

### DISCUSSION THREAD

**Discussion by gttj2010** [ImageMagick/ImageMagick#6490] [Help]
### ImageMagick version

7.1.1-Q16-HDRI

### Operating system, version and so on

Win 10

### Description

Olá,

I've been trying to extract text from a png image for days with ImageMagick, but without success. So far, I managed to "remove" the background and make the text more visible for the tesseract to extract, but it only sees part of it. The best formatting so far that I managed was with the command:

convert "C:\Users\ponto\Downloads\PDF\CR-000001.png" -separate -background black -compose plus -flatten "C:\Users\ponto\Downloads\PDF\CR-000004.png"

From this point, the tesseract already sees something, but I can't change the font size or the color and so on to see if it improves the extraction. I could only apply the morphologic, but when I apply the tesseract I can't see anything.

Follow the original image, with some censorship...

![CR-000001](https://github.com/ImageMagick/ImageMagick/assets/25445846/7a7f061b-ec74-433e-90da-82f35b64a637)

And I get as output

![CR-000004](https://github.com/ImageMagick/ImageMagick/assets/25445846/b43640aa-721d-4123-a6fc-40bd0bebafb8)

**Comment by fmw42** [comment]
Using division normalization with -compose divide and -threshold, I get only something similar (on Mac)

`magick image.png -alpha off -colorspace gray \( +clone -blur 0x3 \) +swap -compose divide -composite -threshold 35% x.png`

![x](https://github.com/ImageMagick/ImageMagick/assets/13458218/c3196447-7a8f-4760-96f9-f1a7cc36fa58)

You can adjust the threshold as desired.

**Reply by gttj2010** [reply]
Hello,

Very Thanks!!

How did you arrive at this reasoning?

**Reply by fmw42** [reply]
Division normalization (with small blur) helps enhance thin lines.  (Normally one uses a large blur to remove gradual background color).  The threshold helps remove lighter regions leaving the text and lines that are more black.

**Reply by fmw42** [reply]
See https://imagemagick.org/Usage/compose/#divide

**Reply by gttj2010** [reply]
Coming back here to leave a note:

I did several tests with the line above changing the threshold from 5% to 5% and some blur variations, both before and after the "x" and none were satisfactory or complementary to get all the fields I needed. The "CPF" and "SFPC" fields were never accessed. The fields "AMPARO LEGAL" and "NAME" were approximate, but not exact. However, it was worth the attempt and the help of the colleague above who made his time available to try to help me.
